pkg/storage/stores/factory.go

```diff
@@ -0,0 +1,29 @@
+package stores
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/aws"
+	"github.com/cortexproject/cortex/pkg/chunk/azure"
+	"github.com/cortexproject/cortex/pkg/chunk/gcp"
+	cortex_local "github.com/cortexproject/cortex/pkg/chunk/local"
+	"github.com/cortexproject/cortex/pkg/chunk/storage"
+)
+
+// NewObjectClient makes a new ObjectClient of the desired type.
+func NewObjectClient(storeType string, cfg storage.Config) (chunk.ObjectClient, error) {
```

Comment:
Why not using https://github.com/cortexproject/cortex/blob/8f59b141ac55e8884436047cddb93498234a11e4/pkg/chunk/storage/factory.go#L280 ?

Reply:
This too needs updating cortex to latest master. I will open a PR to update it.